### PR TITLE
London class 8 - Craig D'Silva - JavaScript Core 1  - week 1

### PR DESCRIPTION
**Pull request by craig-dsilva**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Craig
- Your City: London
- Your Slack Name: Craig - Trainee

# Homework Details

- Module: JavaScript Core 1
- Week: 1

# Notes

- What did you find easy?
 The exercises were easy to understand

- What did you find hard?
 The magic 8 ball

- What do you still not understand?
 I didn't understand what to do for magic 8 ball

- Any other notes?

### CODE REVIEW

File: extra/2-piping.js [hunk 1]
```diff
@@ -16,27 +16,28 @@
...
 
-function format() {
-
+function format(number) {
+  return `£${number.toString()}`;
 }
 
 const startingValue = 2;
 
 // Why can this code be seen as bad practice? Comment your answer.
```
Comment (DomVinyard) [inline extra/2-piping.js:33]: Still need to answer this question (with a comment)

File: extra/2-piping.js [hunk 2]
```diff
@@ -16,27 +16,28 @@
...
-let badCode =
+let badCode = format(multiply(add(startingValue, 10), 2));
 
 /* BETTER PRACTICE */
-
-let goodCode =
-
+let sum = add(startingValue, 10);
+let product = multiply(sum, 2);
+let formatted = `£${product}`;
```
Comment (DomVinyard) [inline extra/2-piping.js:39]: is this going to always produce the correct two decimal places?

File: mandatory/4-tax.js [hunk 3]
```diff
@@ -5,7 +5,9 @@
   Sales tax is 20% of the price of the product.
 */
 
-function calculateSalesTax() {}
+function calculateSalesTax(amount) {
+  return (amount * 20) / 100 + amount;
```
Comment (DomVinyard) [inline mandatory/4-tax.js:9]: this doesn't seem quite right. This function should just calculate the sales tax (not add the sales tax). 

```
return amount * 0.2
```

File: mandatory/4-tax.js [hunk 4]
```diff
@@ -17,7 +19,9 @@ function calculateSalesTax() {}
   Remember that the prices must include the sales tax (hint: you already wrote a function for this!)
 */
 
-function addTaxAndFormatCurrency() {}
+function addTaxAndFormatCurrency(amount) {
+  return `£${calculateSalesTax(amount).toFixed(2)}`;
```
Comment (DomVinyard) [inline mandatory/4-tax.js:23]: this is where we add the sales tax

```
function addTaxAndFormatCurrency(amount) {
  const tax = calculateSalesTax(amount)
  const total = amount + tax
  // etc
}
```

File: extra/2-piping.js (hunk 1 above)
Comment (craig-dsilva) [reply extra/2-piping.js:33]: Sorry, I forgot about this.

File: extra/2-piping.js (hunk 2 above)
Comment (craig-dsilva) [reply extra/2-piping.js:39]: No, but I have added Math.round() to product because the tests provided doesn't want a decimal value.

**Pull request closed without merging**